### ISSUE TITLE
False positive `no-member` when wrapping dataclasses `field`

### ISSUE THREAD

**Issue by Quexington** [pylint-dev/pylint#9972]
### Bug description

```python
If I want to alias/wrap a field for a dataclass, then pylint thinks that the members of the dataclass at that point are actually `Field` objects (at least when checking for `no-member`).  Here's a minimal example:

from dataclasses import dataclass, field

def field_alias(**kwargs):
    # do some stuff analyzing the args

    # The following ignore is "fine" because it only needs to happen in one spot
    return field(**kwargs)  # pylint: disable=invalid-field-call

class ObjectWSubField:
    sub_field: int

@dataclass(frozen=True)
class Temp:
    the_object: ObjectWSubField = field_alias(type=ObjectWSubField)

    def some_method(self):
        # no-member error thrown here and I don't want to throw in an ignore at every spot
        print(self.the_object.sub_field)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
a.py:15:14: E1101: Instance of 'Field' has no 'sub_field' member (no-member)
```
```


### Expected behavior

No error throw because object is correctly identified as the type it is hinted as

### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.12.4 (tags/v3.12.4:8e8a4ba, Jun  6 2024, 19:30:16) [MSC v.1940 64 bit (AMD64)]
```

### OS / Environment

Windows 11

**Comment by alexjircan** [comment]
I have the same issue too :(

**Comment by alexjircan** [comment]
It is similar to #4813
